File: app-backend/src/main/java/com/katya/app/service/AmenityService.java
package com.katya.app.service;

import com.katya.app.dto.response.AmenityResponse;
import com.katya.app.util.enums.Locale;
import com.katya.app.util.enums.PropertyType;

import java.util.List;

public interface AmenityService {

    List<AmenityResponse> getAllAmenities(Locale locale);

    List<AmenityResponse> getAmenitiesForPropertyType(PropertyType propertyType, Locale locale);

    AmenityResponse getAmenityById(Short id, Locale locale);
}

File: app-frontend/src/services/amenityService.ts
// src/services/amenityService.ts
import { publicApi } from '../config/axios'
import { API_ENDPOINTS } from '../config/constants'
import type { Amenity, ApiResponse, Locale, PropertyType } from '../types'

export class AmenityService {
  
  // Get all amenities
  static async getAllAmenities(locale: Locale = 'en'): Promise<Amenity[]> {
    const response = await publicApi.get<ApiResponse<Amenity[]>>(
      `${API_ENDPOINTS.AMENITIES}?locale=${locale}`
    )
    
    return response.data.data
  }

  // Get amenities for specific property type
  static async getAmenitiesForPropertyType(
    propertyType: PropertyType, 
    locale: Locale = 'en'
  ): Promise<Amenity[]> {
    const response = await publicApi.get<ApiResponse<Amenity[]>>(
      `${API_ENDPOINTS.AMENITIES}/property-type/${propertyType}?locale=${locale}`
    )
    
    return response.data.data
  }

  // Get amenity by ID
  static async getAmenityById(id: number, locale: Locale = 'en'): Promise<Amenity> {
    const response = await publicApi.get<ApiResponse<Amenity>>(
      `${API_ENDPOINTS.AMENITIES}/${id}?locale=${locale}`
    )
    
    return response.data.data
  }
}